server/src/test/java/org/apache/kafka/server/ClientMetricsManagerTest.java

```diff
@@ -1023,5 +1177,18 @@ public void testCacheExpirationTaskCancelledOnInstanceUpdate() throws UnknownHos
         assertNotNull(instance);
         assertNotNull(instance.expirationTimerTask());
         assertEquals(1, clientMetricsManager.expirationTimer().size());
+        // Metrics size should remain same on instance update.
+        assertEquals(12, kafkaMetrics.metrics().size());
+        assertEquals((double) 1, getMetric(ClientMetricsManager.ClientMetricsStats.INSTANCE_COUNT).metricValue());
+    }
+
+    private KafkaMetric getMetric(String name) throws Exception {
+        Optional<Entry<MetricName, KafkaMetric>> metric = kafkaMetrics.metrics().entrySet().stream()
```

Comment:
Hmm, why do we need to iterate the whole map instead of doing a `get` based on name on the map?

Reply:
The map has `MetricName` as the key, rather constructing `MetricName` again in tests with tags, I iterated on the map itself. Similar to how we do SelectorTest.java.